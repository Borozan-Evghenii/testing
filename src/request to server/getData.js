const axios = require('axios')
const arrToString = require('../arrToString/arrToString')

const getData = async () => {
	const response = await axios.get('https://jsonplaceholder.typicode.com/users');
	const userId = response.data.map(item => item.id)
	return arrToString(userId)
}


module.exports = getData;