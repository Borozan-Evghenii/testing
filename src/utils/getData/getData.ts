import axios from "axios";
import { arrToString } from "../arrToString/arrToString";

export const getData = async () => {
  try {
    const response = await axios.get(
      "https://jsonplaceholder.typicode.com/users"
    );
    const userId = response.data.map((item: { id: number }) => item.id);
    return arrToString(userId);
  } catch (error) {
    console.log(error);
  }
};
