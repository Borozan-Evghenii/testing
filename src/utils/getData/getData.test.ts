import { beforeAll, describe, expect, test, vi } from "vitest";
import { getData } from "./getData";
import axios from "axios";

vi.mock("axios");

type User = { id: number; name: string; username: string; email: string };

describe("getData ", () => {
  let response: { data: User[] };

  beforeAll(() => {
    response = {
      data: [
        {
          id: 1,
          name: "Leanne Graham",
          username: "Bret",
          email: "Sincere@april.biz",
        },
        {
          id: 2,
          name: "Ervin Howell",
          username: "Antonette",
          email: "Shanna@melissa.tv",
        },
      ],
    };
  });

  test("reques user", async () => {
    axios.get.mockResolvedValue(response);
    const data = await getData();
    expect(axios.get).toBeCalledTimes(1);
    expect(data).toEqual(["1", "2"]);
    expect(data).toMatchSnapshot();
  });
});
