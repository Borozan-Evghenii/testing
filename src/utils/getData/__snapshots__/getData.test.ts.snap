// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`getData  > reques user 1`] = `
[
  "1",
  "2",
]
`;
